RNMail/RNMail.m

```diff
@@ -47,6 +47,40 @@ - (dispatch_queue_t)methodQueue
             [mail setToRecipients:recipients];
         }
         
+        if (options[@"attachment"][@"path"] && options[@"attachment"][@"name"]){
```

Comment:
bail out early. check if attachment is available first and then attachment[path].

Also, If name is optional then why check for it here? Did you mean type?